x/slashing/keeper/signing_info.go

```diff
@@ -141,20 +95,133 @@ func (k Keeper) IsTombstoned(ctx sdk.Context, consAddr sdk.ConsAddress) bool {
+// in the bitmap and the relative bit in that chunk.
+func (k Keeper) GetMissedBlockBitmapValue(ctx sdk.Context, addr sdk.ConsAddress, index int64) (bool, error) {
+	// get the chunk or "word" in the logical bitmap
+	chunkIndex := index / types.MissedBlockBitmapChunkSize
+
+	bs := bitset.New(uint(types.MissedBlockBitmapChunkSize))
+	chunk := k.GetMissedBlockBitmapChunk(ctx, addr, chunkIndex)
+	if chunk != nil {
+		if err := bs.UnmarshalBinary(chunk); err != nil {
+			return false, errors.Wrapf(err, "failed to decode bitmap chunk; index: %d", index)
+		}
+	}
+
+	// get the bit position in the chunk of the logical bitmap
+	bitIndex := index % types.MissedBlockBitmapChunkSize
+	return bs.Test(uint(bitIndex)), nil
+}
+
+// SetMissedBlockBitmapValue sets, i.e. flips, a bit in the validator's missed
+// block bitmap. When missed=true, the bit is set, otherwise it set to zero. The
+// index provided is assumed to be the index in the range [0, SignedBlocksWindow),
+// which represents the bitmap where each bit represents a height, and is
+// determined by the validator's IndexOffset modulo SignedBlocksWindow. This
+// index is used to fetch the chunk in the bitmap and the relative bit in that
+// chunk.
+func (k Keeper) SetMissedBlockBitmapValue(ctx sdk.Context, addr sdk.ConsAddress, index int64, missed bool) error {
+	// get the chunk or "word" in the logical bitmap
+	chunkIndex := index / types.MissedBlockBitmapChunkSize
+
+	bs := bitset.New(uint(types.MissedBlockBitmapChunkSize))
+	chunk := k.GetMissedBlockBitmapChunk(ctx, addr, chunkIndex)
+	if chunk != nil {
+		if err := bs.UnmarshalBinary(chunk); err != nil {
+			return errors.Wrapf(err, "failed to decode bitmap chunk; index: %d", index)
+		}
+	}
+
+	// get the bit position in the chunk of the logical bitmap
+	bitIndex := uint(index % types.MissedBlockBitmapChunkSize)
+	if missed {
+		bs.Set(bitIndex)
+	} else {
+		bs.Clear(bitIndex)
+	}
+
+	updatedChunk, err := bs.MarshalBinary()
+	if err != nil {
+		return errors.Wrapf(err, "failed to encode bitmap chunk; index: %d", index)
+	}
+
+	k.SetMissedBlockBitmapChunk(ctx, addr, chunkIndex, updatedChunk)
+	return nil
+}
+
+// DeleteMissedBlockBitmap removes a validator's missed block bitmap from state.
+func (k Keeper) DeleteMissedBlockBitmap(ctx sdk.Context, addr sdk.ConsAddress) {
+	store := ctx.KVStore(k.storeKey)
+
+	iter := storetypes.KVStorePrefixIterator(store, types.ValidatorMissedBlockBitmapPrefixKey(addr))
 	defer iter.Close()
+
 	for ; iter.Valid(); iter.Next() {
 		store.Delete(iter.Key())
 	}
 }
+
+// IterateMissedBlockBitmap iterates over a validator's signed blocks window
+// bitmap and performs a callback function on each index, i.e. block height, in
+// the range [0, SignedBlocksWindow).
+//
+// Note: A callback will only be executed over all bitmap chunks that exist in
+// state.
+func (k Keeper) IterateMissedBlockBitmap(ctx sdk.Context, addr sdk.ConsAddress, cb func(index int64, missed bool) (stop bool)) {
```

Comment:
What would be a use case for the callback here? Could we just return a slice of MissedBlock directly?